src/thread/ThreadPreview.tsx

```diff
@@ -0,0 +1,52 @@
+import React from "react";
+
+const HiddenThread: React.FC<HiddenThreadProps> = ({ 
+    threadId, 
+    boardId, 
+    threadHidden, 
+    setHideCallback 
+}) => {
+  return (
+    <div className="post hidden" key={threadId}>
+      This thread was hidden{" "}
+      <a
+        href="#"
+        onClick={(e) => {
+          setHideCallback({
+            threadId: threadId,
+            boardId: boardId,
+            hide: threadHidden,
+          });
+          e.preventDefault();
+        }}
+      >
+        [unhide]
+      </a>
+      <style jsx>{`
+        .post.hidden {
+          margin: 0 auto;
+          max-width: 500px;
+          width: calc(100% - 40px);
+          background-color: gray;
+          padding: 20px;
+          border: 1px dashed black;
+          border-radius: 15px;
+        }
+      `}</style>
+    </div>
+  );
+};
+
+export interface HiddenThreadProps {
+  threadId: string;
```

Comment:
You don't actually need the IDs of the thread and board here. Generally we keep these details out of the components library itself. The frontend's callback will contain the information on which thread is being hidden (see my comment there).

Reply:
I'm a little confused - do you mean that I should pass a callback like `() => setHiddenThread({threadId, boardId, hide})` from the front end? so `HiddenThreadProps` would simply be `{ onThreadHidden : <callback function> }`?

Reply:
I kind of got `LinkWithAction` to work, by creating it on the component side and passing in what the frontend sent. Like this:
```  
  let link : LinkWithAction = { 
    onClick: () => onThreadHidden(!hidden), 
  }
  return (
    <div className="post hidden">
      This thread was hidden{" "}
      <ActionLink link={link} allowDefault={false}>
        ...
```
That being said, it's not difficult to change it to what you suggested, if this is some unholy abomination against good practices 😛

Reply:
Also I do have a question about the callback function itself:
> const linkCallback = { 
  onClick: (hide: boolean) => hide ? hideThread({threadId, boardId}) : showThread({threadId, boardId}),
 };
I've been using the `useSetThreadHidden` hook. It seeeeeems to do the same thing, but is there a reason why you would use one over the other?
Also, should the `hidden` boolean passed to the `HiddenThread` component be the current state of the thread (hidden, so `true`) or what we want to pass to the callback (we want to unhide the thread, so `false`)? Admittedly, this feels like semantics, since it's very simple to change.